src/main/java/com/conveyal/r5/analyst/DataPreloader.java

```diff
@@ -0,0 +1,239 @@
+         * static sites) find paths to all the points in the extents specified in the task.
+         *
+         * That is to say, the extents specified in the task are the destination points of a single point request,
+         * are both the origin and destination points of a static site, and are only the origins of a regional
+         * accessibility task, where the destinations are given by the specified opportunity grid.
+         */
+        public Key (AnalysisTask task) {
+            this.networkId = task.graphId;
+            // A supplied scenario ID will always override any full scenario that is present.
+            this.scenarioId = task.scenarioId != null ? task.scenarioId : task.scenario.id;
+            // We need to link for all of access modes, egress modes, and direct modes (depending on whether transit is used).
+            // See code in TravelTimeComputer for when each is used.
+            this.modes = EnumSet.of(
+                    LegMode.getDominantStreetMode(task.directModes),
+                    LegMode.getDominantStreetMode(task.accessModes),
+                    LegMode.getDominantStreetMode(task.egressModes));
+
+            if (task.isHighPriority() || task.makeStaticSite) {
+                // High Priority is an obsolete term for "single point task".
+                // For single point tasks and static sites, there is no opportunity grid. The grid of destinations is
+                // the extents given in the task, which for static sites is also the grid of origins.
+                this.webMercatorExtents = WebMercatorExtents.forTask(task);
+            } else {
+                // A non-static-site regional task. We expect a valid grid of opportunities to be specified as the
+                // destinations. This is necessary to compute accessibility. So we extract those bounds from the grids.
+                this.webMercatorExtents = WebMercatorExtents.forGrid(((RegionalTask)task).gridData);
+            }
+
+            /**
+             * Travel time results may be combined with many different grids, so we don't want to limit their geographic extent
+             * to that of any one grid. Instead we use the extents supplied in the request.
+             * The UI only sends these if the user has changed them to something other than "full region".
+             * If "full region" is selected, the UI sends nothing and the backend fills in the bounds of the region.
+             *
+             * FIXME: the request bounds indicate either origin bounds or destination bounds depending on the request type.
+             * We need to specify these separately as we merge all the request types.
+             *
+             * Reuse linkages in the base gridPointSet stored in the TransportNetwork as to avoid relinking
+             *
+             * Regional analyses use the extents of the destination opportunity grids as their destination extents.
```

Comment:
Yes, we should probably add more distinct fields for origin extents and destination extents. This could be aided by the new com.conveyal.r5.analyst.WebMercatorExtents class.